### ISSUE TITLE
Support for self closing tags (HTML / XML)

### ISSUE THREAD

**Issue by flochtililoch** [thibaudcolas/curlylint#84]
### Describe the bug

Self closing tags which are valid HTML/XML are throwing off the linter.

### Which terms did you search for in the documentation and issue tracker?

self closing tag

### Environment

- curlylint, version 0.13.0
- macOS Big Sur 11.4

### Steps to reproduce

(Write your steps here:)

1. First, create a file named `my.xml` with the following content
    ```xml
    <foo>
      <bar />
    </foo>
    ```
2. Then, run the linter with the following command
    ```
    curlylint my.xml --rule 'indent: 2'
    ```

### Expected behavior

```
All done! ✨ 🍰 ✨
```

### Actual behavior

```
my.xml
1:7	Parse error: expected one of '>', 'attribute', '{#', '{%', '{{' at 1:7	parse_error
```

To be noted, if you change the file content to:
```xml
<foo>
  <bar></bar>
</foo>
```
the linter won't report any error (as expected).

**Comment by flochtililoch** [comment]
After diving into the code base, I see that self closing tags are supported / implemented, but that they are whitelisted, mostly to support SVGs and the few valid HTML ones. I wonder if if would be beneficial to allow adding more to the whitelist by configuration, or just having a config flag that allows any XML tag to be self closing.

**Comment by williamjmorenor** [comment]
I think I hit this same issue



```
       <div class="text-center">
            <p>{{ cursos.total }} cursos disponibles.
                <br>
                {% if cursos.has_prev %}
                <a class="link-dark" href="{{ url_for("home", page=cursos.prev_num) }}" <i
                    class="bi bi-arrow-left-short" aria-hidden="true"></i>
                    Anterior
                </a>
                {% endif %}
                Página {{ cursos.page }} de {{ cursos.pages }}
                {% if cursos.has_next %}
                <a class="link-dark" href="{{ url_for("home", page=cursos.next_num) }}">Siguiente <i
                        class="bi bi-arrow-right-short" aria-hidden="true"></i></a>
                {% endif %}
            </p>
        </div>
```

This report a error but the sintax is ok

```
←[4mnow_lms\templates\inicio\mooc.html←[0m
←[2m120:88←[0m  Parse error: expected one of '>', 'attribute', '{#', '{%', '{{' at 120:88       parse_error
```

And render as expected

![image](https://user-images.githubusercontent.com/3522386/139538669-083096bd-532d-4242-be9f-ef282c107d1c.png)

I want to run curlylint as part of CI but I can not fix this issue with <br></br> because this is not valid html

**Comment by flochtililoch** [comment]
>  I think I hit this same issue

From the snippet you pasted, there's a missing `>` line 5:
```
...cursos.prev_num) }}" <i
```
should be
```
...cursos.prev_num) }}"><i
```

**Comment by adamchainz** [comment]
curlylint describes itself as an HTML linter. I don't think it's intended that you can lint XML files with it. That is quite a different problem since XML has many other features...
